Lib/fontTools/feaLib/builder.py

```diff
@@ -1658,38 +1658,31 @@ def makeVariablePos(self, location, varscalar):
 
         return default, device
 
+    def makeAnchorPos(self, varscalar, deviceTable, location):
+        device = None
+        if not isinstance(varscalar, VariableScalar):
+            if deviceTable is not None:
+                device = otl.buildDevice(dict(deviceTable))
+            return varscalar, device
+        default, device = self.makeVariablePos(location, varscalar)
+        if device is not None and deviceTable is not None:
+            raise FeatureLibError(
+                "Can't define a device coordinate and variable scalar", location
+            )
+        return default, device
+
     def makeOpenTypeAnchor(self, location, anchor):
         """ast.Anchor --> otTables.Anchor"""
         if anchor is None:
             return None
-        variable = False
         deviceX, deviceY = None, None
         if anchor.xDeviceTable is not None:
             deviceX = otl.buildDevice(dict(anchor.xDeviceTable))
         if anchor.yDeviceTable is not None:
             deviceY = otl.buildDevice(dict(anchor.yDeviceTable))
-        for dim in ("x", "y"):
-            varscalar = getattr(anchor, dim)
-            if not isinstance(varscalar, VariableScalar):
-                continue
-            if getattr(anchor, dim + "DeviceTable") is not None:
-                raise FeatureLibError(
-                    "Can't define a device coordinate and variable scalar", location
-                )
-            default, device = self.makeVariablePos(location, varscalar)
-            setattr(anchor, dim, default)
-            if device is not None:
-                if dim == "x":
-                    deviceX = device
-                else:
-                    deviceY = device
-                variable = True
-
-        otlanchor = otl.buildAnchor(
-            anchor.x, anchor.y, anchor.contourpoint, deviceX, deviceY
-        )
-        if variable:
-            otlanchor.Format = 3
```

Comment:
This checks was redundant, since `otl.buildAnchor()` will set format to 3 if either `deviceX` or `deviceY` is not None, so I took the opportunity to simplify the code.